core/drivers/crypto/crypto_api/acipher/rsa.c

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2020 NXP
+ *
+ * Brief   Crypto RSA interface implementation to enable HW driver.
+ */
+#include <drvcrypt.h>
+#include <crypto/crypto.h>
+#include <tee_api_defines_extensions.h>
+#include <tee/tee_cryp_utl.h>
+#include <utee_defines.h>
+
+#include "local.h"
+
+TEE_Result crypto_acipher_alloc_rsa_keypair(struct rsa_keypair *key,
+					    size_t size_bits)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+
+	struct drvcrypt_rsa *rsa = NULL;
+
+	if (!key || !size_bits) {
+		CRYPTO_TRACE("Parameters error (key @%p) (size %zu bits)", key,
+			     size_bits);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	rsa = drvcrypt_get_ops(CRYPTO_RSA);
```

Comment:
I believe adding this requires a bit of rework. For now, `rsa.c` is built only if the HW RSA driver is implemented. We fully agree that software fallback is something we need. But if you agree, that's something we would like to do it a second phase ; especially since this PR is expected by @sahilnxp

Reply:
If it's planned it OK to save it for later.